.github/workflows/test.yml

```diff
@@ -0,0 +1,60 @@
+name: Test
+
+on: push
+
+env:
+  COMPOSE_FILE: docker-compose.test.yml
+  DOCKER_TOKEN: ${{ secrets.DOCKER_TOKEN }}
+  DOCKER_IMAGE: ${{ github.repository }}
+  DOCKER_REGISTRY: ghcr.io
+  DOCKER_USERNAME: ${{ github.repository_owner }}
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Branch Tag Action
+        uses: nimblehq/branch-tag-action@v1.1
+
+      - name: Docker login to Github container registry
+        run: "echo $DOCKER_TOKEN | docker login $DOCKER_REGISTRY -u $DOCKER_USERNAME --password-stdin"
+
+      - name: Docker pull
+        if: env.BRANCH_TAG != 'latest' && env.BRANCH_TAG != 'development'
+        run: | 
+          echo Pulling built image for the branch
+          docker-compose pull test || true
+
+      - name: Docker build
+        run: |
+          bin/docker-prepare
+          docker-compose build
+          docker-compose push test
+
+  tests:
+    name: Tests
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Branch Tag Action
+        uses: nimblehq/branch-tag-action@v1.1
+
+      - name: Docker login to Github container registry
```

Comment:
As we can set it via the `$DOCKER_REGISTRY` variable, should it be "Docker container registry" instead?

Reply:
Make sense, fixed.